src/servers/src/mysql/writer.rs

```diff
@@ -134,10 +136,13 @@ impl<'a, W: AsyncWrite + Unpin> MysqlResultWriter<'a, W> {
     ) -> Result<()> {
         match create_mysql_column_def(&query_result.schema) {
             Ok(column_def) => {
+                let tracing_span = Span::current();
                 // The RowWriter's lifetime is bound to `column_def` thus we can't use finish_one()
                 // to return a new QueryResultWriter.
                 let mut row_writer = writer.start(&column_def).await?;
-                while let Some(record_batch) = query_result.stream.next().await {
+                while let Some(record_batch) =
+                    query_result.stream.next().trace(tracing_span.clone()).await
```

Comment:
in_current_span?

Reply:
Good catch 👍  It turns out this `.trace()` is not necessary. I've removed it.